hack/externalnode/install-vm.sh

```diff
@@ -141,17 +153,278 @@ EOF
+        docker_download_url="https://download.docker.com/linux/centos/docker-ce.repo"
+        cmd="sudo yum-config-manager --add-repo ${docker_download_url}"
+        execute_command "$cmd" 1m
+        docker_install_packages="docker-ce docker-ce-cli containerd.io docker-compose-plugin"
+        cmd="sudo yum install -y ${docker_install_packages}"
+        execute_command "$cmd" 1m
+        cmd="sudo systemctl start docker"
+        execute_command "$cmd" 1m
+        ;;
+    esac
+}
+
+# Install all pre-requisite packages needed for install.
+install_prerequisite_packages() {
+    set +eo pipefail
+    # Install docker, docker-compose only when running services in containers.
+    docker version
+    if [[ "$?" -ne 0 ]]; then
+        install_container_runtime
+    fi
+    # Install docker-compose
+    docker-compose version
+    if [ "$?" -ne 0 ]; then
+        cmd="curl -L "${DOCKER_COMPOSE}-$(uname -s)-$(uname -m)" -o ${DOCKER_COMPOSE_DIR}"
+        execute_command "$cmd" 1m
+        if [ "$?" -eq 0 ]; then
+            echo "docker-compose is downloaded successfully"
+            chmod +x ${DOCKER_COMPOSE_DIR}
+            ln -s ${DOCKER_COMPOSE_DIR} /usr/bin/docker-compose
+        else
+            echoerr "Error docker-compose couldn't be downloaded, exiting"
+            exit 2
+        fi
+    fi
+    set -eo pipefail
+}
+
+# Create docker-compose file with ovs and antrea-agent container specific parameters.
+create_docker_compose_file() {
+    cat << EOF > $AGENT_CONF_PATH/docker-compose.yaml
+version: "3"
+
+services:
+  ovs:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: ovs
+    volumes:
+        - /var/log/openvswitch:/var/log/openvswitch
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /lib/modules:/lib/modules
+    command: $OVS_BIN
+    network_mode: host
+    cap_add:
+        - NET_ADMIN
+        - SYS_NICE
+        - SYS_ADMIN
+        - IPC_LOCK
+
+  antrea-agent:
+    image: $ANTREA_DOCKER_IMAGE
+    container_name: antrea-agent
+    volumes:
+        - /etc/antrea/:/etc/antrea
+        - /var/run/antrea:/var/run/antrea
+        - /var/run/openvswitch:/var/run/openvswitch
+        - /var/lib/cni:/var/lib/cni
+        - /var/log/antrea:/var/log/antrea
+        - /proc:/host/proc
+        - /var/run/netns:/host/var/run/netns
+        - /run/xtables.lock:/run/xtables.lock
+    environment:
+        - NODE_NAME=$NODE_NAME
+    command: $ANTREA_AGENT_BIN --config $AGENT_CONF_PATH/antrea-agent.conf --logtostderr=false --log_file=$AGENT_LOG_DIR/antrea-agent.log
+    network_mode: host
+    privileged: true
+    cap_add:
+        - SYS_MODULE
+EOF
+}
+
+create_antrea_agent_service_file() {
+    cat >$AGENT_CONF_PATH/antrea-agent.sh << 'EOF'
+#!/bin/bash
+
+AGENT_CONF_PATH="/etc/antrea"
+AGENT_LOG_DIR="/var/log/antrea"
+PIDFILE="$AGENT_CONF_PATH/antrea-agent.pid"
+DOCKER_COMPOSE_YAML="$AGENT_CONF_PATH/docker-compose.yaml"
+COUNT=0
+LIMIT=1000
+
+declare -a OVS_KERNEL_MODULES=("vport_geneve" "openvswitch")
```

Comment:
vport_geneve is not needed with VM Agent case?

Reply:
Can you tell

Reply:
vport_geneve is used for OVS tunnel, but we do not use tunnel in VM Agent case. So I don't think it is required. Please help verify if it could be removed. If yes, let's remove it.

Reply:
Okay